core/resource/service.go

```diff
@@ -89,6 +89,10 @@ func (s *Service) Update(ctx context.Context, r *domain.Resource) error {
 		return err
 	}
 
+	// Details["_metadata"] is not allowed to be updated by users
+	// value for this field should only set by the provider on FetchResources
+	delete(r.Details, "_metadata")
```

Comment:
what happens if `r.Details` is nil or empty ?

Reply:
golang takes care of it already, I've tried to add `if r.Details != nil` wrapper and golang gives warning that it's an unnecessary checking